radix-engine/src/engine/kernel.rs

```diff
@@ -717,6 +717,14 @@ where
 
         Ok(output)
     }
+
+    pub fn finalize(&mut self) -> Result<(), RuntimeError> {
```

Comment:
Can we integrate this with `track` finalize and also make this function be a consuming `self` one?

Reply:
Done. I've moved track ownership to kernel and `Kernel.finalize` now outputs a TrackReceipt. I could figure out how to make the modules `on_finished_processing` consume `self` as well, not sure if that's even possible because of them being `Box<dyn >>` with statically unknown size.